hknweb/events/views.py

```diff
@@ -28,11 +49,11 @@ def show_details(request, id):
         'event': event,
         'rsvp': rsvp,
     }
-    if request.user.is_authenticated:
-        return render(request, 'events/show_details.html', context)
-    return redirect('/events')
+    return render(request, 'events/show_details.html', context)
 
 @csrf_exempt  #doing this for now bc idk how to make csrf work
+@login_required(login_url='/accounts/login/')
+@check_account_access
```

Comment:
How would you feel about using a more granular permission that can be granted to multiple groups, like:
```py
from django.contrib.auth.decorators import permission_required

@permission_required('events.add_rsvp', login_url='/accounts/login')
def my_view(request):
    ...
```
https://docs.djangoproject.com/en/2.2/topics/auth/default/#the-permission-required-decorator

Each model ('Bar') has the following [default permissions](https://docs.djangoproject.com/en/2.2/topics/auth/default/#default-permissions) generated:

```
add: user.has_perm('foo.add_bar')
change: user.has_perm('foo.change_bar')
delete: user.has_perm('foo.delete_bar')
view: user.has_perm('foo.view_bar')
```

This would have the benefit of not having to add groups to `@check_account_access` as we add groups, as long as we grant this permission to each group.
As a side effect, this requires only one decorator instead of two.

Also, the [`settings.LOGIN_URL`](https://docs.djangoproject.com/en/2.2/ref/settings/#std:setting-LOGIN_URL) appears to set the value of `login_url`, which furthermore has default value `/accounts/login/`, so this parameter is actually unnecessary in our project:

```py
@permission_required('events.add_rsvp')
def my_view(request):
    ...
```

Reply:
If we did this, we would need to add the appropriate permissions to the candidate and officer group, which is easier to organize, but more work to set up

Reply:
Is there a way to indicate what permissions each group should have, and even automatically add them upon startup or something

Reply:
The way to do that would be a data migration, listed in https://github.com/compserv/hknweb/issues/94:

https://docs.djangoproject.com/en/2.2/topics/migrations/#data-migrations

This would allow us to create a database migration that adds data, instead of modifying the tables.

Reply:
How about I merge this for now, then create a separate pull request to refactor this?